src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub/files/jupyterhub/03-profiles.py

```diff
@@ -522,38 +520,27 @@ def render_profile(
             base_profile_extra_mounts(),
             configure_user(username, groups),
             configure_user_provisioned_repositories(username),
-            profile_kubespawner_override,
+            profile.get("kubespawner_override"),
         ],
         {},
     )
 
-    # We need to merge any env vars from the spawner with any overrides from the profile
-    # This is mainly to ensure JUPYTERHUB_ANYONE/GROUP is passed through from the spawner
-    # to control dashboard access.
-    envvars_fixed = {**(profile["kubespawner_override"].get("environment", {}))}
-
-    def preserve_envvars(spawner):
-        # This adds in JUPYTERHUB_ANYONE/GROUP rather than overwrite all env vars,
```

Comment:
The behavior of Kubespawner changed in v5 so kubespawner_override's default behavior is merge instead of replace now so I'm now simplifing this part.
reference: https://jupyterhub-kubespawner.readthedocs.io/en/latest/changelog.html#id24